packages/wrangler/src/assets.ts

```diff
@@ -201,17 +202,19 @@ export const syncAssets = async (
 			}
 		};
 		// add to queue and run it if we haven't reached concurrency limit
-		void queue.add(() =>
-			doUpload().then((res) => {
-				completionJwt = res.jwt || completionJwt;
-			})
+		queuePromises.push(
+			queue.add(() =>
+				doUpload().then((res) => {
+					completionJwt = res.jwt || completionJwt;
+				})
+			)
 		);
 	}
 	queue.on("error", (error) => {
 		logger.error(error.message);
 		throw error;
 	});
-	await queue.onIdle();
+	await Promise.all(queuePromises);
```

Comment:
Could you put a comment in here explaining why we are not using `onIdle()` to prevent someone from "optimizing" it back and then having to go down the same rabbit hole you did.

Reply:
Done!